samcli/lib/sync/flows/generic_api_sync_flow.py

```diff
@@ -67,6 +68,8 @@ def __init__(
     def gather_resources(self) -> None:
         self._definition_uri = self._get_definition_file(self._api_identifier)
         self._swagger_body = self._process_definition_file()
+        if self._swagger_body:
```

Comment:
What happens if `self._swagger_body` is `None`, are we skipping this sync flow if that is the case?

Reply:
This is only for mypy compliance for Optional type, we do not allow none type for this (this is only None if the definition file is None), as we will throw exception during sync